velox/type/Type.h

```diff
@@ -507,6 +542,96 @@ class TypeBase : public Type {
   }
 };
 
+/*
+ * This class represents the Decimal Type to represent the fixed-point numbers
+ * in Velox. The parameter "precision" represents the number of digits the
+ * Decimal Type can support and scale represents the number of digits to the
+ * right of the decimal point.
+ */
+template <TypeKind KIND>
+class DecimalType : public Type {
+ public:
+  static constexpr uint8_t kMaxPrecision =
+      KIND == TypeKind::SHORT_DECIMAL ? 18 : 38;
+  static constexpr uint8_t kMaxScale =
+      KIND == TypeKind::SHORT_DECIMAL ? 18 : 38;
+  static_assert(
+      KIND == TypeKind::SHORT_DECIMAL || KIND == TypeKind::LONG_DECIMAL);
+
+  DecimalType(const uint8_t precision = 18, const uint8_t scale = 0)
+      : Type(KIND), precision_(precision), scale_(scale) {
+    VELOX_CHECK_LE(scale, precision);
+    VELOX_CHECK_LE(precision, kMaxPrecision);
+  }
+
+  inline bool isPrimitiveType() const override {
+    return true;
+  }
+
+  inline bool isFixedWidth() const override {
+    return true;
+  }
+
+  inline const char* kindName() const override {
+    return TypeTraits<KIND>::name;
+  }
+
+  inline uint32_t size() const override {
+    // DECIMAL TypeSignature can have 2 children which can be
```

Comment:
In the context of "size" method, precision and scale are not children. Only types can be children. This method should return 0.

Reply:
> @karteekmurthys Thank you for iterating on this PR. Some quick comments. What's left to allow queries like `SELECT decimal '1.2'` to succeed?

To add support for decimal literals the Prestissimo must be able to interpret DECIMAL tokens. This requires adding new tokens to presto_cpp/main/types/TypeSignature.g4 and generate the parsers through ANTRL.
Reference PR: https://github.com/facebookexternal/presto_cpp/pull/714